File: Data Structures & Algorithms/validate-parentheses/submission-0.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> c;
        for(char i : s){
            if(i == '(' || i == '{'|| i == '['){
                c.push(i);
            }else{
                if(c.empty()){
                    return false;
                }
                else if(i==')' && c.top() != '('){
                    return false;
                }
                else if(i=='}' && c.top() != '{'){
                    return false;
                }
                else if(i==']' && c.top() != '['){
                    return false;
                }
                c.pop();
            }
        }
        return c.empty();
    }
};

File: Data Structures & Algorithms/simplify-path/submission-0.java
class Solution {
    public String simplifyPath(String path) {
        Stack<String> stack = new Stack<>();
        String[] parts = path.split("/");
        for(String s : parts){
            if(s.equals("") || s.equals(".")){
                continue;
            }
            else if(s.equals("..")){
                if(!stack.isEmpty()){
                    stack.pop();
                }
            }
            else{
                stack.push(s);
            }
        }
        StringBuilder result = new StringBuilder();
        for(String dir : stack){
            result.append("/").append(dir);
        }
        if(result.length() == 0){
            return "/";
        }
        return result.toString();
    }
}